tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -14161,6 +14161,97 @@ static bool IsUsageAttributeCompatible(AttributeList::Kind kind, bool &usageIn,
   return true;
 }
 
+bool DiagnoseRayQueryObject(QualType qt, Declarator &D, Sema &S) {
+  // now we know that qt is a TemplateSpecializationType
+  // handle diagnostics that relate to enumtypedecl vs literal first
+  const TemplateSpecializationType *TST =
+      dyn_cast<TemplateSpecializationType>(qt);
+  if (!TST) {
+    return false;
+  }
```

Comment:
```suggestion
  if (!TST) 
    return false;
```

https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements

(repeated throughout this code)